core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/sync/EccNodesSync.java

```diff
@@ -0,0 +1,141 @@
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilder;
+import com.ericsson.bss.cassandra.ecchronos.connection.StatementDecorator;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+
+public final class EccNodesSync
+{
+    private static final Integer DEFAULT_CONNECTION_DELAY_IN_MINUTES = 30;
+    private static final String COLUMN_ECCHRONOS_ID = "ecchronos_id";
+    private static final String COLUMN_DC_NAME = "datacenter_name";
+    private static final String COLUMN_NODE_ID = "node_id";
+    private static final String COLUMN_NODE_ENDPOINT = "node_endpoint";
+    private static final String COLUMN_NODE_STATUS = "node_status";
+    private static final String COLUMN_LAST_CONNECTION = "last_connection";
+    private static final String COLUMN_NEXT_CONNECTION = "next_connection";
+    private static final String ECCHORONS_ID_PRE_STRING = "ecchronos-";
+    private static final  String KEYSPACE_NAME = "ecchronos";
+    private static final  String TABLE_NAME = "nodes_sync";
+    private final CqlSession mySession;
+    private final PreparedStatement myInsertStatement;
+    private final StatementDecorator myStatementDecorator;
+
+    private EccNodesSync(final Builder builder)
+    {
+        this.mySession = Preconditions.checkNotNull(builder.mySession, "Session cannot be null");
+        this.myStatementDecorator = Preconditions
+                .checkNotNull(builder.myStatementDecorator, "StatementDecorator cannot be null");
+        this.myInsertStatement = mySession.prepare(QueryBuilder.insertInto(KEYSPACE_NAME, TABLE_NAME)
+                .value(COLUMN_ECCHRONOS_ID, bindMarker())
+                .value(COLUMN_DC_NAME, bindMarker())
+                .value(COLUMN_NODE_ENDPOINT, bindMarker())
+                .value(COLUMN_NODE_STATUS, bindMarker())
+                .value(COLUMN_LAST_CONNECTION, bindMarker())
+                .value(COLUMN_NEXT_CONNECTION, bindMarker())
+                .value(COLUMN_NODE_ID, bindMarker())
+                .build()
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM));
+    }
+
+    public ResultSet acquireNode(final String endpoint, final Node node) throws UnknownHostException
+    {
+        String ecchronosID = ECCHORONS_ID_PRE_STRING.concat(InetAddress.getLocalHost().getHostName());
+        return insertNodeInfo(ecchronosID, node.getDatacenter(), endpoint,
+                node.getState().toString(), Instant.now(),
+                Instant.now().plus(DEFAULT_CONNECTION_DELAY_IN_MINUTES, ChronoUnit.MINUTES),
+                node.getHostId());
+    }
+
+    @VisibleForTesting
```

Comment:
I'd say that you can let it as public, as we'd use this method to populate initial nodes state on cassandra